### ISSUE TITLE
Hosting view hierarchy for platform view : 0 Using PlatformView on Android with Kotlin. What is platform view : 0??

### ISSUE THREAD

**Issue by surajsahani** [flutter/flutter#121873]
## Steps to Reproduce

1. using PlatformView for showing my Android Native View From third party Native android SDK.
2. Currently UI is integrated, but it's not consistent, UI not always shows, here is logs please check.

```
I/PlatformViewsController: Hosting view in view hierarchy for platform view: 0
E/FrameEvents: updateAcquireFence: Did not find frame.

```
**Expected results:** 

```dart
```

</details>

<details>
  <summary>Logs</summary>
I/okhttp.OkHttpClient: <-- END HTTP (1398-byte body)
I/PlatformViewsController: Hosting view in view hierarchy for platform view: 0
D/Surface: lockHardwareCanvas
E/OpenGLRenderer: fbcNotifyFrameComplete error: undefined symbol: fbcNotifyFrameComplete
E/OpenGLRenderer: fbcNotifyNoRender error: undefined symbol: fbcNotifyNoRender
E/FrameEvents: updateAcquireFence: Did not find frame.
D/Surface: lockHardwareCanvas
E/FrameEvents: updateAcquireFence: Did not find frame.
D/DecorView[]: onWindowFocusChanged hasWindowFocus true

```
```

```
```

```
```

</details>

**Comment by huycozy** [comment]
Can you please elaborate on this by providing a GitHub repository for a complete example code that contains Flutter and Platform (Android) code? 
Also, please provide the output of `flutter doctor -v` as well. Thank you!

**Comment by surajsahani** [comment]
```
 flutter doctor -v
[✓] Flutter (Channel stable, 3.7.1, on Ubuntu 22.04.2 LTS 5.19.0-35-generic, locale en_IN)
    • Flutter version 3.7.1 on channel stable at /home/martial/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7048ed95a5 (5 weeks ago), 2023-02-01 09:07:31 -0800
    • Engine revision 800594f1f4
    • Dart version 2.19.1
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /home/martial/Android/Sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /snap/android-studio/125/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[!] Android Studio (version 2022.1)
    • Android Studio at /home/martial/Documents/android-studio
    • Flutter plugin version 72.0.2
    • Dart plugin version 221.6096
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2021.3)
    • Android Studio at /snap/android-studio/125/android-studio
    • Flutter plugin version 72.1.1
    • Dart plugin version 213.7433
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Connected device (3 available)
    • 22041216I (mobile) • HAXOVGYPCMQSXSZ5 • android-arm64  • Android 12 (API 31)
    • Linux (desktop)    • linux            • linux-x64      • Ubuntu 22.04.2 LTS 5.19.0-35-generic
    • Chrome (web)       • chrome           • web-javascript • Google Chrome 110.0.5481.177

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```

**Comment by surajsahani** [comment]
> Can you please elaborate on this by providing a GitHub repository for a complete example code that contains Flutter and Platform (Android) code? Also, please provide the output of `flutter doctor -v` as well. Thank you!

```
 flutter doctor -v
[✓] Flutter (Channel stable, 3.7.1, on Ubuntu 22.04.2 LTS 5.19.0-35-generic, locale en_IN)
    • Flutter version 3.7.1 on channel stable at /home/martial/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7048ed95a5 (5 weeks ago), 2023-02-01 09:07:31 -0800
    • Engine revision 800594f1f4
    • Dart version 2.19.1
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /home/martial/Android/Sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /snap/android-studio/125/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[!] Android Studio (version 2022.1)
    • Android Studio at /home/martial/Documents/android-studio
    • Flutter plugin version 72.0.2
    • Dart plugin version 221.6096
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2021.3)
    • Android Studio at /snap/android-studio/125/android-studio
    • Flutter plugin version 72.1.1
    • Dart plugin version 213.7433
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Connected device (3 available)
    • 22041216I (mobile) • HAXOVGYPCMQSXSZ5 • android-arm64  • Android 12 (API 31)
    • Linux (desktop)    • linux            • linux-x64      • Ubuntu 22.04.2 LTS 5.19.0-35-generic
    • Chrome (web)       • chrome           • web-javascript • Google Chrome 110.0.5481.177

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```

**Comment by huycozy** [comment]
Please provide a complete example code as well. You may also try my [example](https://github.com/huycozy/platformviews-demo) and compare with yours code to check the cause of the issue.

**Comment by surajsahani** [comment]
This issue i raised to check what is value 0.
I open the project and check in log there's is 
**PlatformViewsController com.example.test_platformviews       I  Hosting view in view hierarchy for platform view: 0**

So i want to know what is view: 0 here?.

**Comment by surajsahani** [comment]
Steps to reproduce.

https://github.com/surajsahani/flutter-aar-demo.git

1. Use this link for my project demo.
2. Now install and open app, there's currently one button which will render correctly for first time.
3. Now kill the app and reopen application again, and UI will not show 2nd time.

**Comment by huycozy** [comment]
I see you are using `com.lib.loopsdk.core.customViews.TriggerButton` on your sample code (activity_main.xml), it comes from 3rd library. And it isn't shown on the layout even the first time. Can you confirm it works on Android native project? 

When I add a normal Button, it's displaying well:

```xml
<Button
    android:id="@+id/button_first"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:text="@string/next"
    app:layout_constraintEnd_toEndOf="parent"
    app:layout_constraintStart_toStartOf="parent"
    app:layout_constraintTop_toTopOf="parent" />
```

**Comment by surajsahani** [comment]
> 

Yes it works in Native Code, here's complete source code, please check

https://github.com/surajsahani/Native-SDK-demo.git

Also attaching video for native code

https://user-images.githubusercontent.com/22853459/223324843-f9783351-4e17-4636-9169-6cfbd947a102.mp4

**Comment by huycozy** [comment]
Thanks for providing the native project. I run it and can see the Android native project works normally. 

But I still can't see that button displayed on all 3 of my devices when running Flutter project. What's your device information (OS, model)?

Are there any corresponding remote repository dependencies of these dependencies that you can use?

```
    implementation files('libs/logger-debug.aar')
    implementation files('libs/fct-loop-sdk-debug.aar')
```

Also, please retry this on the latest stable and master channels to check if it still persists or not.

**Comment by surajsahani** [comment]
This is my testing device information.
Device - Redmik50i
Specs below.
![redmi k50i](https://user-images.githubusercontent.com/22853459/223408161-8cc6a562-df7e-4b13-bd65-2580eb9281c6.jpeg)

I am already in stable channel attachment below
![flutter-version](https://user-images.githubusercontent.com/22853459/223408285-90ce988d-b0d5-4f1f-886f-3551a1105d74.jpeg)

However i will try again with master channel

**Comment by surajsahani** [comment]
After switching to master channel, issue still not fixed.

![Screenshot from 2023-03-07 17-22-52](https://user-images.githubusercontent.com/22853459/223414674-bea66842-0db7-4687-957c-c252c61a770a.png)

Demo video of master channel attaching below

https://user-images.githubusercontent.com/22853459/223414953-be9183ff-a49a-427d-9523-adc4c12317ab.mp4

**Comment by huycozy** [comment]
I can see that button is shown but need to run with `profile` mode. And the issue doesn't happen even when bringing app to the background and back to the foreground (as steps you did in the video).

Verified on `Realme 6, Android 11`, profile mode.

<details>
<summary>Demo</summary>


https://user-images.githubusercontent.com/104349824/223614065-fe96dab5-ffee-4301-b82e-d866d6e3eade.mp4


</details>

Please retry and confirm if it also works for you on profile mode.

**Comment by surajsahani** [comment]
In Profile mode also View is not be rendered after reopening app.


https://user-images.githubusercontent.com/22853459/223653062-0d011a44-782d-4165-b0a7-8563a2237ae5.mp4



this behavior should be same in all mode like **debug** **profile**, **release** because i need  release mode for further development process.

**Comment by surajsahani** [comment]
Sure let me try again with other devices

**Comment by huycozy** [comment]
I see that you also have another issue here https://github.com/flutter/flutter/issues/121973 and it seems to be similar to this one. Duplicate issues should be grouped into a single issue, so everyone can focus on identifying the problem. 

Also, since the issue can be reproducible on another device as well (demo at https://github.com/flutter/flutter/issues/121973#issuecomment-1459655819) and it was labeled, I will close this issue in favor of  #121973. If there are any different points that I miss here, please write in the comments and I will reopen this for further discussion.
